src/controllers/students/student.controller.js

```diff
@@ -0,0 +1,160 @@
+/* 
+
+-> import student.service from services/student folder
+-> there is an object of studentModel which has different functions :-
+1. addStudent - to add the student in the student array
+2. fetchAllStudents - to get all the students present in the student array
+3. fetchStudentById - pass an id , and fetch the data of that particular student
+4. removeStudentById - pass and id, and delete the student with that particular id
+5. updateStudentById - pass an id, and update the details of the student at that particular id
+    NOTE :- YOU CANNOT CHANGE THE ID OF THE STUDENT
+
+*/
+
+const studentModel = require('../../services/student/student.services');
+
+const studentController = {
+    addStudent: (req, res) => { // addstudent function of studentController object
+
+        const students = studentModel.getAllStudents(); 
+        const id = students.length > 0 ? Math.max(...students.map(s => s.id)) + 1 : 1; // finding the max id from student model
+    
+        let { name, address, gender, course_name } = req.body;
+    
+        // Check if student with this auto-generated ID already exists
+        if (studentModel.studentExists(id)) {
+            return res.status(400).json({ status: 400, error: 'Student with this ID already exists' });
+        }
+    
+        // Trim the input fields
+        name = name.trim();
+        address = address.trim();
+        gender = gender.trim();
+        course_name = course_name.trim();
```

Comment:
do we need trim here ??

Reply:
the trim here will remove all the extra spaces entered by the user